src/components/question-set/EditQSModal.tsx

```diff
@@ -0,0 +1,80 @@
+import React from "react"
+import { Modal, Box, Typography, TextField, Button } from "@mui/material"
+import { QuestionSet } from "../../features/question-set/questionSetSlice"
+
+interface EditQuestionSetModalProps {
+  open: boolean
+  questionSet: QuestionSet | null
+  onClose: () => void
+  onSave: (updatedQuestionSet: QuestionSet) => void
+}
+
+const EditQuestionSetModal: React.FC<EditQuestionSetModalProps> = ({
+  open,
+  questionSet,
+  onClose,
+  onSave,
+}) => {
+  const [title, setTitle] = React.useState("")
+  const [description, setDescription] = React.useState("")
+
+  React.useEffect(() => {
+    if (questionSet) {
+      setTitle(questionSet.title)
+      setDescription(questionSet.description)
+    }
+  }, [questionSet])
+
+  const handleSave = () => {
+    if (questionSet) {
+      onSave({
+        ...questionSet,
+        title,
+        description,
+      })
+      onClose()
+    }
+  }
+
+  return (
+    <Modal open={open} onClose={onClose}>
+      <Box
+        sx={{
+          position: "absolute",
+          top: "50%",
+          left: "50%",
+          transform: "translate(-50%, -50%)",
+          bgcolor: "white",
+          boxShadow: 24,
+          p: 4,
+          borderRadius: 4,
+          minWidth: 300,
+        }}
```

Comment:
Why not css modules?
Also we should avoid centering items using position absolute when possible.